edifice/_component.py

```diff
@@ -274,12 +278,12 @@ class Component(object):
     whenever you have a list of children.
     """
 
-    _render_changes_context = None
-    _render_unwind_context = None
-    _ignored_variables = set()
-    _edifice_internal_parent: 'Component' = None
-    _controller = None
-    _edifice_internal_references = None
+    _render_changes_context: dict | None = None
+    _render_unwind_context: dict | None = None
+    _ignored_variables: set[tp.Text] | None = None
+    _edifice_internal_parent: tp.Optional["Component"] = None
+    _controller: _Controller | None = None
```

Comment:
The type of `_controller` is always an `App`.

You have created `_Controller` which is a kind of abstract superclass for `App` which requires the `_request_rerender` method.

https://github.com/pyedifice/pyedifice/blob/1e55aa5e06e1740078b883a3470b4387734fbc1a/edifice/app.py#L197

Why not make `_Controller` a real superclass for `App`?

Or why not just use the `App` type? Because of the module circular definitions? In the case I would rather refactor the modules and pull some stuff out into a `types.py` file or something. Or at least put a comment on `_Controller` which says **"""TODO This type should not exist it's just an alias for App to break the module cycle."""**